lib/reverse_sentence.rb

```diff
@@ -1,4 +1,58 @@
 # A method to reverse the words in a sentence, in place.
+
+# time complexity is O(1)
```

Comment:
The space complexity and it's reasoning is correct. However the time complexity of your algorithm is not O(1). To figure out the time complexity, you want to count the number of steps or the number of operations your algorithm takes in relation to the size of the input. If the input has n characters and you are looking to examine each character, then just to examine each character, you will need n steps. In this case, you want to add up the number of steps taken by word_reverse and reverse_words to compute the time complexity.